src/test.ts

```diff
@@ -1205,6 +1205,600 @@ describe('YAML Title', () => {
+      - Title
+    key3: value3
+    linter-yaml-title-alias: Title
+    ---
+    # Title
+    `;
+
+    expect(rulesDict['yaml-title-alias'].apply(before, {'YAML aliases new section style': 'Multi-line array'})).toBe(after);
+  });
+
+  it('Does not add alias that matches the filename for multi-line array style aliases section', () => {
+    const before = dedent`
+    ---
+    aliases:
+      - alias1
+    ---
+    # Filename
+    `;
+
+    const after = dedent`
+    ---
+    aliases:
+      - alias1
+    ---
+    # Filename
+    `;
+
+    expect(rulesDict['yaml-title-alias'].apply(before, {'Keep alias that matches the filename': false, 'metadata: file name': 'Filename'})).toBe(after);
+  });
+
+  it('Does not add alias that matches the filename for single-line array style aliases section', () => {
+    const before = dedent`
+    ---
+    aliases: [alias1]
+    ---
+    # Filename
+    `;
+
+    const after = dedent`
+    ---
+    aliases: [alias1]
+    ---
+    # Filename
+    `;
+
+    expect(rulesDict['yaml-title-alias'].apply(before, {'Keep alias that matches the filename': false, 'metadata: file name': 'Filename'})).toBe(after);
+  });
+
+  it('Does not add alias that matches the filename for single string style aliases section', () => {
+    const before = dedent`
+    ---
+    aliases: alias1
+    ---
+    # Filename
+    `;
+
+    const after = dedent`
+    ---
+    aliases: alias1
+    ---
+    # Filename
+    `;
+
+    expect(rulesDict['yaml-title-alias'].apply(before, {'Keep alias that matches the filename': false, 'metadata: file name': 'Filename'})).toBe(after);
+  });
+
+  it('Does not add alias that matches the filename for multi-line array style aliases section, removes previous alias', () => {
+    const before = dedent`
+    ---
+    aliases:
+      - alias1
+      - alias2
+    linter-yaml-title-alias: alias1
+    ---
+    # Filename
+    `;
+
+    const after = dedent`
+    ---
+    aliases:
+      - alias2
+    ---
+    # Filename
+    `;
+
+    expect(rulesDict['yaml-title-alias'].apply(before, {'Keep alias that matches the filename': false, 'metadata: file name': 'Filename'})).toBe(after);
+  });
+
+  it('Does not add alias that matches the filename for single-line array style aliases section, removes previous alias', () => {
+    const before = dedent`
+    ---
+    aliases: [alias1, alias2]
+    linter-yaml-title-alias: alias1
+    ---
+    # Filename
+    `;
+
+    const after = dedent`
+    ---
+    aliases: [alias2]
+    ---
+    # Filename
+    `;
+
+    expect(rulesDict['yaml-title-alias'].apply(before, {'Keep alias that matches the filename': false, 'metadata: file name': 'Filename'})).toBe(after);
+  });
+
+  it('Does not add alias that matches the filename for single string style aliases section, removes previous alias', () => {
+    const before = dedent`
+    ---
+    aliases: alias1
+    linter-yaml-title-alias: alias1
+    ---
+    # Filename
+    `;
+
+    const after = dedent`
+    ---
+    aliases:
```

Comment:
Should it remove the key as well rather than leaving it empty when dealing with a single string?